packages/gallery/src/components/item/itemHelper.js

```diff
@@ -0,0 +1,73 @@
+import { window, utils, isSiteMode, isSEOMode } from 'pro-gallery-lib';
+
+function shouldChangeActiveElement() {
+  return (isSiteMode() || isSEOMode()) && !utils.isMobile() && window.document;
+}
+
+export function onAnchorFocus({
+  itemContainer,
+  shouldUseExperimentalFeature,
+  itemAnchor,
+}) {
+  if (shouldChangeActiveElement() && shouldUseExperimentalFeature) {
+    /* Relevant only for Screen-Reader cases:
+         When we navigate on the accessibility tree, screen readers stops and focuses on the <a> tag,
+         so it will not go deeper to the item-container keyDown event */
+    const activeElement = window.document.activeElement;
+    if (activeElement === itemAnchor) {
+      itemContainer.focus();
+    }
+  }
+}
+
+export function changeActiveElementIfNeeded({
+  prevProps,
+  currentProps,
+  itemContainer,
+}) {
+  try {
+    if (
+      shouldChangeActiveElement() &&
+      window.document.activeElement.className
+    ) {
+      const activeElement = window.document.activeElement;
+
+      //check if focus is on 'gallery-item-container' in current gallery
+      const isThisGalleryItemInFocus = () =>
+        !!window.document.querySelector(
+          `#pro-gallery-${currentProps.domId} #${String(activeElement.id)}`
+        );
+      const isGalleryItemInFocus = () =>
+        String(activeElement.className).indexOf('gallery-item-container') >= 0;
+      //check if focus is on 'load-more' in current gallery
+      const isThisGalleryShowMoreInFocus = () =>
```

Comment:
I dont get the "ThisGallery" part. this code runs on that same gallery.

Reply:
I understand I noticed this too, I did not want to make changes that are not part of an issue I came to solve,
But I agree with you, now I own this code, so I'll look and see and improve what's right.